modules/common/Utils.py

```diff
@@ -88,3 +106,31 @@ def resource_for_stage(resource):
         resource_stage.uri = resource.uri
         resource_stage.output_filename = os.path.basename(resource.uri)
         return resource_stage
+
+
+class CustomSubProcException(Exception):
+    """Custom subprocess exception"""
+
+    pass
+
+
+def subproc(cmd: str, **kwargs) -> None:
+    """Run a cmd as a subprocess. This catches errors that are 'raised' in
+    the STDERR even if the command doesn't return a code > 0.
+
+    Arguments:
+        cmd -- command string to run
+
+    Raises:
+        CustomSubProcException
+    """
+    try:
+        sp = subprocess.run(
+            cmd, shell=True, text=True, capture_output=True, check=False, **kwargs
+        )
+        sp.check_returncode()
+    except subprocess.CalledProcessError as err:
+        raise CustomSubProcException(err) from err
```

Comment:
It would be nice to have this method capturing the error in the logs with the command information as shown in the comment below, otherwise, we can't really pinpoint to what the error is about in the log messages.
I think it makes sense that the information logged in here is similar to the comment below and we think about a different level of details to be logged in the caller code, or maybe it just goes back to the client of that caller code, i.e. the client of *run_riot()*

Reply:
The same message logged by this _helper function_ can be raised in an exception to the caller